compiler/lsp-tests/src/Daml/Lsp/Test/Util.hs

```diff
@@ -0,0 +1,62 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE OverloadedStrings #-}
+module Daml.Lsp.Test.Util
+    ( Cursor
+    , cursorPosition
+    , expectDiagnostics
+    , damlId
+    , replaceDoc
+    ) where
+
+import Control.Applicative.Combinators
+import Control.Lens
+import Control.Monad
+import Control.Monad.IO.Class
+import qualified Data.Text as T
+import Language.Haskell.LSP.Test hiding (message)
+import qualified Language.Haskell.LSP.Test as LspTest
+import Language.Haskell.LSP.Types
+import Language.Haskell.LSP.Types.Lens
+import Test.Tasty.HUnit
+
+import DA.Test.Util
+
+-- | Convenient grouping of 0-based line number, 0-based column number.
+type Cursor = (Int, Int)
+
+cursorPosition :: Cursor -> Position
+cursorPosition (line,  col) = Position line col
+
+searchDiagnostics :: (DiagnosticSeverity, Cursor, T.Text) -> PublishDiagnosticsParams -> Assertion
```

Comment:
I'd call this requireDiagnostic, it's not really a "search" in a meaningful way.

Reply:
I just copied the name from `daml-ghc-shake-test-ci`, happy to change it but I expect that this will probably change quite a bit anyway so this whole function might disappear.